plugins/inputs/azure_monitor/azure_monitor.go

```diff
@@ -0,0 +1,856 @@
+      aggregation = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+    
+    # defines the resources to collect metrics from
+    [[inputs.azure_monitor.resource_group_target.resource]]
+      resource_type = "<<RESOURCE_TYPE>>"
+      metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+      aggregation = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+      
+  # resource group target #2 to collect metrics from resources under it with resource type
+  [[inputs.azure_monitor.resource_group_target]]
+    resource_group = "<<RESOURCE_GROUP_NAME>>"
+
+    [[inputs.azure_monitor.resource_group_target.resource]]
+      resource_type = "<<RESOURCE_TYPE>>"
+      metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+      aggregation = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+  
+  # subscription target #1 to collect metrics from resources under it with resource type    
+  [[inputs.azure_monitor.subscription_target]]
+    resource_type = "<<RESOURCE_TYPE>>"
+    metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+    aggregation = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+    
+  # subscription target #2 to collect metrics from resources under it with resource type    
+  [[inputs.azure_monitor.subscription_target]]
+    resource_type = "<<RESOURCE_TYPE>>"
+    metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+    aggregation = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+`
+
+func (am *AzureMonitor) Description() string {
+	return "Gather Azure resources metrics from Azure Monitor API"
+}
+
+func (am *AzureMonitor) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init is for setup, and validating config.
+func (am *AzureMonitor) Init() error {
+	err := am.checkConfigValidation()
+
+	if err != nil {
+		return err
+	}
+
+	err = am.getAccessToken()
+
+	if err != nil {
+		return err
+	}
+
+	err = am.createResourceGroupTargetsFromSubscriptionTargets()
+
+	if err != nil {
+		return err
+	}
+
+	err = am.createResourceTargetsFromResourceGroupTargets()
+
+	if err != nil {
+		return err
+	}
+
+	err = am.getResourceTargetsMetrics()
+
+	if err != nil {
+		return err
+	}
+
+	am.setResourceTargetsAggregation()
+	am.checkTargetsMaximumMetrics()
+
+	am.Log.Info("Total targets: ", len(am.ResourceTargets))
+
+	return nil
+}
+
+func (am *AzureMonitor) checkConfigValidation() error {
+	if am.SubscriptionID == "" {
+		return fmt.Errorf("subscription_id is empty or missing. Please check your configuration")
+	}
+
+	if am.ClientID == "" {
+		return fmt.Errorf("client_id is empty or missing. Please check your configuration")
+	}
+
+	if am.ClientSecret == "" {
+		return fmt.Errorf("client_secret is empty or missing. Please check your configuration")
+	}
+
+	if am.TenantID == "" {
+		return fmt.Errorf("tenant_id is empty or missing. Please check your configuration")
+	}
+
+	if len(am.ResourceTargets) == 0 && len(am.ResourceGroupTargets) == 0 && len(am.SubscriptionTargets) == 0 {
+		return fmt.Errorf("there is no target to collect metrics from. Please check your configuration")
```

Comment:
Maybe shorten the error message to 
```suggestion
		return fmt.Errorf("no target to collect metrics from in your configuration")
```